app/classifier/tasks/highlighter/index.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Markdown } from 'markdownz';
+import GenericTaskEditor from '../generic-editor';
+import GenericTask from '../generic';
+import LabelEditor from './label-editor';
+import LabelRenderer from './label-renderer';
+import HighlighterSummary from './summary';
+import TextRenderer from '../../annotation-renderer/text';
+
+export default class Highlighter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.createButtons = this.createButtons.bind(this);
+    this.createLabelAnnotation = this.createLabelAnnotation.bind(this);
+    this.selectableArea = this.selectableArea.bind(this);
+  }
+
+  handleChange(toolIndex, e) {
+    let selection = document.getSelection();
+    this.createLabelAnnotation(selection, toolIndex);
+  }
+
+  createLabelAnnotation(selection, toolIndex) {
+    // currently we only deal with one selection at a time
+    const range = selection.getRangeAt(0);
+    const offset = this.getOffset(selection);
+    const start = offset + range.startOffset;
+    const end = offset + range.endOffset;
+    const task = this.props.workflow.tasks[this.props.annotation.task];
+    const labelInformation = task.highlighterLabels[toolIndex];
+    const selectable = this.selectableArea(selection, range, offset, start, end);
+    if (selectable) {
+      const newAnnotation = Object.assign({}, this.props.annotation, { _toolIndex: toolIndex });
+      newAnnotation.value.push({
+        labelInformation: labelInformation,
+        start: start,
+        end: end,
+        text: range.toString()
+      });
+      this.props.onChange(newAnnotation);
+    }
+    selection.collapseToEnd();
+  }
+
+  selectableArea(selection, range, offset, start, end){
+    const spansNodes = range.startContainer !== range.endContainer
+    const noAreaSelected = start === end;
+    const subjectSelection = selection.anchorNode.parentElement.parentElement.className === 'label-renderer';
+    const isSelectable = !spansNodes && !noAreaSelected && subjectSelection;
+    return isSelectable;
+  }
+
+  getOffset(selection) {
+    const { anchorNode } = selection;
+    const { parentNode } = anchorNode;
+    let start = 0;
+    for (const node of parentNode.childNodes) {
```

Comment:
I addressed this and another IE issue in #3873. I tried to replace `for...of` with `forEach`, but I couldn't find an equivalent to the `break` behavior using `forEach`. So, I replaced `forEach` with a simple `for` loop.